src/amp/lib/edition.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Determine if a string is one of the permitted edition strings
+ *
+ * @param s The string to test
+ */
+export function isEdition(s: string): s is Edition {
+	return ['UK', 'US', 'INT', 'AU'].includes(s);
```

Comment:
If we were to use an `Enum` for Edition in [index.d.ts#L160][enum], we could make this [more independent][ts].

```ts
enum Edition {
    UK = 'UK',
    US = 'US',
    INT = 'INT',
    AU = 'AU',
}

/**
 * Determine if a string is one of the permitted edition strings
 *
 * @param s The string to test
 */
export function isEdition(s: string): s is Edition {
	return s in Edition;
}

const validEdition = ['A','B', 'C', 'UK', 'INT'].filter(isEdition);
```

Maybe one for a subsequent PR, though, as it might have an impact on Dotcom.

[ts]: https://www.typescriptlang.org/play?#code/KYOwrgtgBAogJgSwC4IPYigbwLACgoFQCqA0lALxQDkpVANHocQMoXVHP2OECSAcgBU2VfgK75CAQSLDp4gL548AegBUqxqqgARYEmAAnCAhDAoCAGZQAhlADOSAyYDm5u1HRnUVpAAszAA6Gxkj6cFDAiCjo9o4udpqaUAACAdYG1tDuAv6xTiCuSKhQ+g6aynjAAB4BqAZIUBZgIADG0RgIdvDIaCAAFHYAXHkuAJTD7p2wUb1YeACQBnpgBhiTGN3tANx4irh4LegOUABu1gA2CHCbs5QA2lSS9FQAQvTUAMLvNCTfolQAXQAdBYEOd9AY+p0buhRlsgA
[enum]: https://github.com/guardian/dotcom-rendering/blob/5386ee021368b8aaf45bf52d86a6b792125d7000/index.d.ts#L160

Reply:
Maybe `as const` is better, actually: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions ?

Reply:
> You could force a local geo class on the body?

That's a good point but it looks like we currently don't support AMP elements in Storybook (we'd probably need to use an [addon like this](https://storybook.js.org/addons/storybook-amp). This feels a bit out of scope for this PR but may be useful to install it in a separate PR.

Reply:
Very good point, I just had a play around with changing it to an enum and it looks like enums don't behave exactly as expected when declared in .d.ts files, I think possibly another one for another PR.